pkg/kubectl/history.go

```diff
@@ -305,6 +305,32 @@ func applyHistory(ds *extensionsv1beta1.DaemonSet, history *appsv1beta1.Controll
 	return clone, nil
 }
 
+// controlledSSHistories returns all ControllerRevisions controlled by the given StatefulSet
+func controlledSSHistories(apps clientappsv1beta1.AppsV1beta1Interface, namespace, name string) (*appsv1beta1.StatefulSet, []*appsv1beta1.ControllerRevision, error) {
```

Comment:
Using a `runtime.Object` as the returned value and doing conversions at the calling site would be one way to make this polymorphic.